File: src/cpp/11_napisy_wprowadzenie/zad09.cpp
/*
ZAD-09 — Rozdziel informacje o pracowniku

**Poziom:** ★☆☆
**Tagi:** `split`, `formatowanie`

### Treść

Wczytaj linię z danymi pracownika rozdzielonymi średnikami `;`:

1. Imię, 2) Nazwisko, 3) Miejsce urodzenia, 4) Stanowisko, 5) Zarobki

Wypisz każdą informację w osobnej linii z etykietą.

### Wejście

* 1. linia: dane w formacie `Imię; Nazwisko; Miasto; Zawód; Zarobki;`

### Wyjście

Pięć linii w formacie:

* `Imię: ...`
* `Nazwisko: ...`
* `Miejsce urodzenia: ...`
* `Zawód: ...`
* `Zarobki: ...`

### Przykład

**Wejście:**

```
Jan; Kowalski; Warszawa; Programista; 1000;
```

**Wyjście:**

```
Imię: Jan
Nazwisko: Kowalski
Miejsce urodzenia: Warszawa
Zawód: Programista
Zarobki: 1000
```

### Uwagi

* Po `split(';')` usuń ewentualne spacje z brzegów pól (np. `strip()`).
* Ostatni średnik może powodować pusty element na końcu — zignoruj go.

*/
#include <cassert>
#include <iostream>
#include <sstream>
#include <vector>

std::vector<std::string> dane(std::string &napis) {
  std::vector<std::string> wynik;
  std::stringstream ss(napis);

  for (std::string slowo; ss >> slowo;) {
    if (slowo.size() > 1) {
      wynik.push_back(slowo.substr(0, slowo.size() - 1));
      if (ss.peek() == ';') ss.ignore();
    }
  }

  return wynik;
}

void wypiszDane(std::string &napis) {
  std::vector<std::string> wynik = dane(napis);
  assert(wynik.size() == 5);

  std::cout << "Imie: " << wynik[0] << std::endl;
  std::cout << "Nazwisko: " << wynik[1] << std::endl;
  std::cout << "Miejsce urodzenia: " << wynik[2] << std::endl;
  std::cout << "Zawod: " << wynik[3] << std::endl;
  std::cout << "Zarobki: " << wynik[4] << std::endl;
}

int main() {
  std::string napis = "Jan; Kowalski; Warszawa; Programista; 1000";
  wypiszDane(napis);
  return 0;
}

File: src/js/11_napisy_wprowadzenie/zad09.js
/*
ZAD-09 — Rozdziel informacje o pracowniku

**Poziom:** ★☆☆
**Tagi:** `split`, `formatowanie`

### Treść

Wczytaj linię z danymi pracownika rozdzielonymi średnikami `;`:

1. Imię, 2) Nazwisko, 3) Miejsce urodzenia, 4) Stanowisko, 5) Zarobki

Wypisz każdą informację w osobnej linii z etykietą.

### Wejście

* 1. linia: dane w formacie `Imię; Nazwisko; Miasto; Zawód; Zarobki;`

### Wyjście

Pięć linii w formacie:

* `Imię: ...`
* `Nazwisko: ...`
* `Miejsce urodzenia: ...`
* `Zawód: ...`
* `Zarobki: ...`

### Przykład

**Wejście:**

```
Jan; Kowalski; Warszawa; Programista; 1000;
```

**Wyjście:**

```
Imię: Jan
Nazwisko: Kowalski
Miejsce urodzenia: Warszawa
Zawód: Programista
Zarobki: 1000
```

### Uwagi

* Po `split(';')` usuń ewentualne spacje z brzegów pól (np. `strip()`).
* Ostatni średnik może powodować pusty element na końcu — zignoruj go.

*/

// Funkcja rozdziela informacje o pracowniku zapisane w jednej linii
// Złożoność czasowa: O(n), gdzie n to długość stringu rekord
// Złożoność pamięciowa: O(n) dla przechowania tablicy danych i wyniku
function rozdzielRekord(rekord) {
  let dane = rekord.split(";");
  let imie = dane[0].trim();
  let nazwisko = dane[1].trim();
  let miejsceUrodzenia = dane[2].trim();
  let zawod = dane[3].trim();
  let zarobki = dane[4].trim();

  let lista = [
    "Imię: " + imie,
    "Nazwisko: " + nazwisko,
    "Miejsce urodzenia: " + miejsceUrodzenia,
    "Zawód: " + zawod,
    "Zarobki: " + zarobki,
  ];
  return lista;
}

// Test

function test() {
  let input = "Jan; Kowalski; Warszawa; Programista; 1000;";
  let expectedOutput = [
    "Imię: Jan",
    "Nazwisko: Kowalski",
    "Miejsce urodzenia: Warszawa",
    "Zawód: Programista",
    "Zarobki: 1000",
  ];
  let output = rozdzielRekord(input);

  console.assert(
    output.length === expectedOutput.length &&
      output.every((element, index) => element === expectedOutput[index]),
    "Test nie powiodl sie"
  );
  console.log("Test przeszedl pomyslnie");
}

test();

